lib/helper/perfherderDataQueryHelper.py

```diff
@@ -0,0 +1,121 @@
+                        return_result['browser_type_list'].append(browser_type)
+                    if machine_platform not in return_result['machine_platform_list']:
+                        return_result['machine_platform_list'].append(machine_platform)
+            with open(PerfherderDataQueryHelper.DEFAULT_HASAL_SIGNATURES, "w+") as fh:
+                json.dump(return_result, fh)
+        return return_result
+
+    @staticmethod
+    def filter_signatures_by_whitelist(input_signatures_data, input_white_list):
+        if len(input_white_list) > 0:
+            for pattern in input_white_list:
+                logger.debug('Filter Black word by pattern: {}'.format(pattern))
+                for sig_key, sig_obj in input_signatures_data['signature_data'].items():
+                    suite_name = sig_obj.get('suite_name', '')
+                    if re.match(pattern, suite_name):
+                        logger.debug('Del sig [{sig}] with suite name [{suite_name}]'.format(sig=sig_key, suite_name=suite_name))
+                        del input_signatures_data['signature_data'][sig_key]
+        return input_signatures_data
+
+    @staticmethod
+    def convert_query_result(input_json, signature_data, input_result_dict):
+        """
+        generate a list which contains the result from input_json and signature_data between begin and end date.
+        @param input_json:
+        @param signature_data:
+        @return: a list which contains result
+        """
+        for sig in input_json:
+            for data in input_json[sig]:
+                suite_name_full = signature_data['signature_data'][sig]['suite_name'].split()
+                suite_name = suite_name_full[0]
+                push_tiemstamp = data['push_timestamp']
+                perfherder_data_index = "%s:%s:%s" % (suite_name, signature_data['signature_data'][sig]['browser_type'], signature_data['signature_data'][sig]['machine_platform'])
+                if push_tiemstamp in input_result_dict:
+                    if perfherder_data_index in input_result_dict[push_tiemstamp]['perfherder_data']:
+                        input_result_dict[push_tiemstamp]['perfherder_data'][perfherder_data_index]['value'].append(data['value'])
+                    else:
+                        input_result_dict[push_tiemstamp]['perfherder_data'][perfherder_data_index] = {"value": [data['value']],
+                                                                                                       "signature": sig}
+                else:
+                    input_result_dict[push_tiemstamp] = {'perfherder_data': {perfherder_data_index: {"value": [data['value']],
+                                                                                                     "signature": sig}}}
+        return input_result_dict
+
+    @staticmethod
+    def get_perfherder_data(input_white_list, input_query_days=14, input_query_channel="mozilla-central", input_hasal_framework_no=9):
```

Comment:
we need the comments to describe the input parameters, like the format of white list.

Reply:
already updated